### PR TITLE
🐛 fix: sidebar topic spinner, ordering and optimistic-row glitches around new-topic sends

### PR DESCRIPTION
**Pull request by tjx666**
#### 💻 Change Type

- [ ] ✨ feat
- [x] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 👷 build
- [ ] ⚡️ perf
- [ ] ✅ test
- [ ] 📝 docs
- [ ] 🔨 chore

#### 🔗 Related Issue

Regression introduced by #16745 (fix 2). Related to #16576 investigation.

#### 🔀 Description of Change

Fixes for the "send from the new-topic view" flow and the sidebar topic-list glitches surfaced by the same investigation:

**1. Skip `saveToTopic` when a send is still in flight in the new-topic view**

Clicking "New Topic" while a send from the new-topic view was still running archived the optimistic `tmp_*` messages into a spurious "Default Topic" and raced the in-flight topic creation. The running send owns topic creation, so `openNewTopicOrSaveTopic` now early-returns when `operationSelectors.isInputLoadingByContext` reports an in-flight send in the `_new` context. The blocked window only lasts until the real topic is created and auto-switched to.

To make the blocked window visible instead of a silent no-op click, a shared `topicSelectors.isNewTopicSendInFlight` selector grays out the sidebar "New Topic" nav item and disables the command-menu "New Topic" item while it holds; the action guard backstops the hotkey path. `NavItem`'s `disabled` prop previously had no visual state at all, so it now dims (opacity 0.5 + not-allowed cursor) for permission-based disables too.

Along the way the chat-input `SaveTopic` action-bar component was deleted outright: it was registered in `actionMap` but no current layout references the `saveTopic` key — dead code from the legacy chat input layout (the Cmd+N hotkey and command menu call `openNewTopicOrSaveTopic` directly and are unaffected).

**2. Release the migrated optimistic-topic loading owner (regression from #16745)**

The first send takes a `topicLoadingIds` owner for the optimistic `tmp_*` topic, and `internal_replaceTopicId` migrates that owner (ref count) onto the real topic id. #16745 converged the run spinner onto the persisted `status === 'running'` and removed the transports' run-end `internal_updateTopicLoading(id, false)` calls — which also removed the only release for this migrated creation owner. Result: after any run that creates a topic via gateway/hetero, the sidebar spinner sticks forever (until page reload clears the in-memory count).

Fix: release the creation owner right after the optimistic topic resolves, in both the gateway and hetero branches of `sendMessageInConversation`.

**3. Status-only topic updates no longer bump `updatedAt` (sidebar jumping)**

The `updateTopic` reducer stamped `updatedAt: new Date()` on every field update, including pure `status` flips (run-end `unread`, mark-read on view). Since the sidebar sorts by `updatedAt`, every run end / topic read yanked that row to the top — and the bump then reverted on the next refetch (the server orders by latest-message time), so rows visibly jumped around. Status-only writes now keep the existing `updatedAt`.

**4. Refetches no longer clobber optimistic sidebar state**

Two races where a topic-list refetch whose server query ran before a local write landed after it:

- A run-end `'unread'` write could be reverted to `'running'` by a stale refetch, leaving the sidebar spinning forever on a finished topic. Optimistic `updateTopicStatus` writes are now tracked in a TTL-bounded (15s) pending map and re-applied over stale fetched rows until the fetch confirms propagation.
- A refetch landing mid-send (e.g. the fire-and-forget `refreshTopic` after a previous run's topic creation or terminal) wholesale-replaced the bucket items and wiped the in-flight `tmp_topic_*` optimistic row, so the new topic was invisible in the sidebar until the server returned the real topicId. Fetch reconciliation now re-prepends surviving `tmp_topic_*` rows — they only leave the bucket via `replaceTopicId` (send resolved) or `deleteTopic` (rollback).

**5. Unread dot during the post-visible-output terminal tail**

After `visible_output_end` the run is still doing terminal bookkeeping (unread persist, title summary) and #16518 intentionally masks the running icon there, which left a blank icon gap until the persisted `'unread'` landed. The sidebar item now shows the unread dot immediately in that window (skipped while the user is viewing the topic, matching `markTopicUnread`'s own guard).

#### 🧪 How to Test

Gateway mode (cloud), from the new-topic view:

1. Send a message → real topic is created and auto-switched to.
2. Wait for the run to finish → the sidebar spinner stops (topic flips `running → unread`/`active`); previously it spun forever.
3. While the send is still in flight (before the real topic appears), click "New Topic" → no spurious "Default Topic" is created; the click is a no-op until the topic resolves.
4. Send message N, wait for the New Topic button to unblock, immediately send N+1 → the optimistic topic row stays visible in the sidebar throughout the send.
5. Let a long run finish in the background → its topic does not jump above topics you started later, and reading an old topic does not reorder it; the row flips straight to the unread dot with no blank-icon gap.

- [x] Tested locally
- [x] Added/updated tests

#### 📝 Additional Information

Also includes an unrelated one-line repo-tooling tweak: `disable-model-invocation` removed from the skills-audit skill frontmatter so agents can invoke it directly.

A regression test now drives the gateway branch with a resolved `topicId` and asserts the migrated `topicLoadingIds` owner is released (verified to fail without the fix).

Known accepted edge case (commented in code): cancelling during phase 1 right after `internal_replaceTopicId` can still leak the migrated owner via the rollback path (it only releases the tmp id). Rare enough to leave out of this PR.

### CODE REVIEW

File: src/store/chat/slices/agentRun/actions/entries/conversationLifecycle.ts [hunk 1]
```diff
@@ -845,6 +845,13 @@ export class ConversationLifecycleActionImpl {
           clearNewKey: true,
           skipRefreshMessage: true,
         });
+        // Same as the gateway branch: resolveOptimisticTopic migrated the
+        // optimistic topic's loading owner onto the real id; the hetero run
+        // spinner is owned by persisted `status === 'running'`, so release the
+        // creation owner here or the sidebar spinner sticks forever.
+        if (optimisticTopic && optimisticTopicResolved) {
+          this.#get().internal_updateTopicLoading(heteroData.topicId, false);
```
[the remark this answers is not in the document]
Comment (tjx666) [reply src/store/chat/slices/agentRun/actions/entries/conversationLifecycle.ts]: Fixed in 0b1580367d — the release is moved into the executor `finally`, so the migrated creation owner is held through the pre-`running` startup gap and released once the run settles (error paths included). It cannot mask `waitingForHuman`: the sidebar item renders that state with higher priority than the running icon. Added a regression test that fails against the previous release timing.

**Comment by tjx666** [comment]
@codex review latest code
